src/main/java/com/getbase/sync/SyncProcess.java

```diff
@@ -40,6 +41,27 @@ public boolean run() {
         return true;
     }
 
+    /**
+     * @param timeToRunInMillis - maximum time to run for sync process
+     * @return true if sync was full - there are no more elements to fetch, false otherwise
+     */
+    public boolean run(final long timeToRunInMillis) {
+        long startTime = currentTimeMillis();
+
+        if (!init()) {
+            return true;
+        }
+
+        while (fetchMore()) {
+            process();
+            if (currentTimeMillis() - startTime >= timeToRunInMillis) {
+                return false;
+            }
+        }
+
+        sessionManager.clearSession(deviceUUID);
```

Comment:
`clearSession()` should be called no matter if sync was full or not (if we only were able to go beyond `init()`). Actually it should be in `finally` block.